### ISSUE TITLE
explore the files inside the torrent before downloading?

### ISSUE THREAD

**Issue by MartinMatomo** [TorrentStream/TorrentStream-Android#84]
Hi,
I have a torrent that contains several files inside, how to explore these files and choose which one to stream ? As an example, the attached image.
![photo_2022-10-29_13-53-07](https://user-images.githubusercontent.com/18522936/198834179-0f01fe65-a8b2-4da6-9df4-62bab9e5eab9.jpg)

**Comment by se-bastiaan** [comment]
What have you tried already? There is a function to get the file names of the files in the torrent metadata: https://github.com/TorrentStream/TorrentStream-Android/blob/develop/library/src/main/java/com/github/se_bastiaan/torrentstream/Torrent.java#L254

**Comment by MartinMatomo** [comment]
Hi, I managed to display the file names in a custom Listview, but I'm looking for how to launch the download by specifying the index of the selected file ? 
I found the line: torrentSession.download(torrentInfo, saveDirectory, null, priorities, null, torrent_flags_t.from_int(0)); in the startStream() method.  How to specify the file to download ?

**Comment by se-bastiaan** [comment]
It's the function above the one I linked before..

**Comment by MartinMatomo** [comment]
Yes, with the function you linked, I managed to display the file names contained in a torrent file, I have the list of all the file names on a ListView, now I'm looking for how to download only the selected file by returning its index on the ListView

**Comment by se-bastiaan** [comment]
I am pointing you to the same place now. Because what you are looking for is in the same file, about 100 lines above it.

**Comment by MartinMatomo** [comment]
Yes, thank you, I had found the solution. Thank you for your help.
